packages/client/hmi-client/src/services/charts.ts

```diff
@@ -508,6 +511,222 @@ export const createForecastChart = (
 	return spec;
 };
 
+export const createForecastChartAnnotation = (axis: 'x' | 'y', datum: number, label: string) => {
+	const layerSpec = {
+		description: `At ${axis} ${datum}, add a label '${label}'.`,
+		encoding: {
+			[axis]: { datum }
+		},
+		layer: [
+			{
+				mark: {
+					type: 'rule',
+					strokeDash: [4, 4]
+				}
+			},
+			{
+				mark: {
+					type: 'text',
+					align: 'left',
+					dx: 5,
+					dy: -5
+				},
+				encoding: {
+					text: { value: label }
+				}
+			}
+		]
+	};
+	const annotation = {
+		id: uuidv4(),
+		layerSpec,
+		isLLMGenerated: false,
+		metadata: { axis, datum, label }
+	};
+	return annotation;
+};
+
+export const generateForecastChartAnnotation = async (
```

Comment:
Actually, if `this` context is not needed, it’s generally safer to use an arrow function because its behaviour is more predictable, as it doesn’t create its own `this`. However, for consistency, I’ll just switch to a normal function since it seems that's what we use in the most cases throughout the app.